lib/api/CAnomalyJobConfig.cc

```diff
@@ -674,6 +676,59 @@ void CAnomalyJobConfig::CDataDescription::parse(const rapidjson::Value& analysis
     m_TimeFormat = parameters[TIME_FORMAT].fallback(EMPTY_STRING); // Ignore
 }
 
+void CAnomalyJobConfig::CAnalysisConfig::parseDetectorsConfig(const rapidjson::Value& detectorsConfig) {
+
+    // The Job config has already been validated by Java before being passed to
+    // the C++ backend. So we can safely assume that the detector config is a
+    // non-null array - hence this check isn't strictly necessary.
+    if (detectorsConfig.IsArray()) {
+        m_Detectors.resize(detectorsConfig.Size());
+        int fallbackDetectorIndex{0};
+        for (std::size_t i = 0; i < detectorsConfig.Size(); ++i) {
+            m_DetectorRules[fallbackDetectorIndex].clear();
+            m_Detectors[i].parse(detectorsConfig[fallbackDetectorIndex], m_RuleFilters,
+                                 (m_SummaryCountFieldName.empty() == false), fallbackDetectorIndex,
+                                 m_DetectorRules[fallbackDetectorIndex]);
+
+            if (m_PerPartitionCategorizationEnabled) {
+                if (m_CategorizationFieldName.empty()) {
+                    throw CAnomalyJobConfigReader::CParseError(
+                        "per_partition_categorization enabled without a categorization field");
+                }
+                if (m_Detectors[i].partitionFieldName().empty()) {
+                    throw CAnomalyJobConfigReader::CParseError(
+                        "per_partition_categorization enabled without a partition field");
+                }
+                if (m_CategorizationPartitionFieldName.empty()) {
+                    m_CategorizationPartitionFieldName = m_Detectors[i].partitionFieldName();
+                } else {
+                    if (m_CategorizationPartitionFieldName !=
+                        m_Detectors[i].partitionFieldName()) {
+                        throw CAnomalyJobConfigReader::CParseError(
+                            "per_partition_categorization enabled when partition "
+                            "field varies between detectors");
+                    }
+                }
+            }
+            ++fallbackDetectorIndex;
+        }
+    }
+}
+
+void CAnomalyJobConfig::CAnalysisConfig::parse() {
```

Comment:
Please give this method a more precise name. It doesn't appear to parse the full analysis config. Maybe something like `reparseDetectorsFromStoredConfig` (or something else that says what it does).